openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/SuggestionRepository.java

```diff
@@ -154,32 +155,32 @@ public void deleteSuggestionInternalForAnEntity(EntityInterface entity) {
 
   @Getter
   public static class SuggestionWorkflow {
-    protected final Suggestion suggestion;
-    protected final MessageParser.EntityLink entityLink;
+    protected final EntityRepository<?> repository;
+    // The workflow is applied to a specific entity at a time
+    protected final EntityInterface entity;
 
-    SuggestionWorkflow(Suggestion suggestion) {
-      this.suggestion = suggestion;
-      this.entityLink = MessageParser.EntityLink.parse(suggestion.getEntityLink());
+    SuggestionWorkflow(EntityRepository<?> repository, EntityInterface entity) {
+      this.repository = repository;
+      this.entity = entity;
```

Comment:
@harshach made the workflow dependant on a repository & entity, rather than at the suggestion level

Reply:
do we need to pass repostory? since its a singleton we can access it anywhere @pmbrull